trinity/rpc/main.py

```diff
@@ -116,7 +117,10 @@ def _get_result(self,
 
             method = self._lookup_method(request['method'])
             params = request.get('params', [])
-            result = method(*params)
+            if asyncio.iscoroutinefunction(method):
```

Comment:
All the RPC methods were synchronous so far. I changed this to support both.

Reply:
I think all JSON-RPC methods should be changed to be async.

Reply:
We can do that but it may look odd if the only async thing in these functions is the `async` keyword, no?

Reply:
Additional question here. If we make all the sync methods `async` but then inside do synchronous operations doesn't that mean we are doing the cardinal sin of blocking the event loop?

Reply:
Either way (async or not) the synchronous block bodies will block the event loop, unless you farm it out to an executor. Not `await`ing here won't change that.

Reply:
Yes, but that is assuming that synchronous and asynchronous functions run in the same thread. We *could* run an event loop in a separate thread (as it was before for the IPCServer) and then I believe all functions that are consumed via `async` / `await` would run on the event loop whereas all others would run blocking in the other thread. However, if the `async` functions do blocking I/O you are still blocking the event loop.